### DISCUSSION TITLE
add_directory Clarification

### DISCUSSION THREAD

**Discussion by msdapptech** [zip-rs/zip-old#313] [Q&A] [answered]
Hello zip-rs team, 

I would like to ask for some clarification on the documentation regarding `ZipWriter.add_directory()`
According to the documentation, it says once you call the method, you can no longer write data to the file. However, from the `write_sample.rs` example, it is clear that you can still add data to the writer. 

I found this confusing. As how I understand it, the "can't write data to the file afterward" means that the specific path passed into `add_directory` as the argument can no longer be used as the name argument for the `start_file`. Is this correct? If it is, I think there might be better way of phrasing it, telling the user that you can still edit more, it is just that the path will forever be treated as a directory, and cannot be used as a file.

Thank you ahead of time, and your library has been helpful.

**Comment by Plecra** [comment, marked as answer]
it's actually just referring to the `io::Write` implementation - internally directories are only represented as normal files with no content, and you're right that this ought to be made clearer in the documentation. I imagine it'd be best to remove that paragraph and give an example of creating a directory containing a couple of files (created an issue #323, no timeline for now, im just shuffling things around for discussions)
